src/transformers/trainer.py

```diff
@@ -3443,7 +3440,8 @@ def _save_tpu(self, output_dir: Optional[str] = None):
     def _save(self, output_dir: Optional[str] = None, state_dict=None):
         # If we are executing this function, we are the process zero, so we don't check for that.
         output_dir = output_dir if output_dir is not None else self.args.output_dir
-        os.makedirs(output_dir, exist_ok=True)
+        if self.args.should_save:
+            os.makedirs(output_dir, exist_ok=True)
```

Comment:
Doesn't this mean the comment above is no longer correct?

Reply:
yes, if _save is called in each rank. the comment above is no longer correct.